src/test/java/address/testutil/ApplicationTest.java

```diff
@@ -0,0 +1,80 @@
+package address.testutil;
+
+import guitests.GuiRobot;
+import javafx.application.Application;
+import javafx.application.HostServices;
+import javafx.application.Preloader;
+import javafx.stage.Stage;
+import org.junit.After;
+import org.junit.Before;
+import org.testfx.api.FxToolkit;
+import org.testfx.api.annotation.Unstable;
+import org.testfx.framework.junit.ApplicationAdapter;
+import org.testfx.framework.junit.ApplicationFixture;
+
+/**
+ * Code mostly from TestFX
+ */
+public abstract class ApplicationTest extends GuiRobot implements ApplicationFixture {
```

Comment:
Instead of this class, we can create something similar to `GuiTestBase` e.g. `GuiUnitTestBase` which implements `ApplicationFixture` for the `start`, `stop` and `init` methods.

This is because we do not need the base class to extend `GuiRobot`/`FxRobot` any more, and we can directly use the `handle` variable in the test class.